continuum/tasks/task_set.py

```diff
@@ -118,19 +121,25 @@ def get_sample(self, index: int) -> np.ndarray:
 
     def __getitem__(self, index: int) -> Tuple[np.ndarray, int, int]:
         """Method used by PyTorch's DataLoaders to query a sample and its target."""
-        sample = self.get_sample(index)
+        x = self.get_sample(index)
         y = self._y[index]
         t = self._t[index]
 
-        if self.data_type != 'text':
-            if self.trsf is not None:
-                sample = self.trsf(sample)
+        if self.data_type == "segmentation":
```

Comment:
ya bcp de if avec l'ajout de la segmentation, je pense que cree un task_set specialis'e pour la segmentation rendrait les choses plus claire / plus propre

Reply:
I'm not sure abouuy a new task set, it's only a few lines added. But I'll make an effort to simply the number of if, you're right

Reply:
I've simplify the task set handling of the different data types.